test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java

```diff
@@ -570,4 +579,17 @@ protected SnapshotInfo getSnapshot(String repository, String snapshot) {
         assertThat(snapshotInfos, hasSize(1));
         return snapshotInfos.get(0);
     }
+
+    protected void awaitMasterFinishRepoOperations() throws Exception {
+        logger.info("--> waiting for master to finish all repo operations on its SNAPSHOT pool");
+        final ThreadPool masterThreadPool = internalCluster().getMasterNodeInstance(ThreadPool.class);
+        assertBusy(() -> {
+            for (ThreadPoolStats.Stats stat : masterThreadPool.stats()) {
+                if (ThreadPool.Names.SNAPSHOT.equals(stat.getName())) {
+                    assertEquals(stat.getActive(), 0);
```

Comment:
The number of active threads relies on `ThreadPoolExecutor#getActiveCount()` which is approximate, but having `core == max` should make this OK I think.